storage/asynchbase/src/main/java/net/opentsdb/storage/Tsdb1xMultiGet.java

```diff
@@ -87,56 +105,54 @@
  * TODO - room for optimization here. E.g. we expect a decent number of
  * TSUIDs. If it's small then we'd potentially have only one or two 
  * GetRequests per batch.
+ * TODO - fix push for reverse.
+ * TODO - fix segment start/end
  * 
  * @since 2.4
  */
-public class Tsdb1xMultiGet implements HBaseExecutor {
+public class Tsdb1xMultiGet implements HBaseExecutor, CloseablePooledObject {
   private static final Logger LOG = LoggerFactory.getLogger(Tsdb1xMultiGet.class);
   
+  /** Reference to the Object pool for this instance. */
+  protected PooledObject pooled_object;
+  
   /** The upstream query node that owns this scanner set. */
-  protected final Tsdb1xHBaseQueryNode node;
+  protected Tsdb1xHBaseQueryNode node;
   
   /** The query config from the node. */
-  protected final TimeSeriesDataSourceConfig source_config;
+  protected TimeSeriesDataSourceConfig source_config;
   
   /** The list of TSUIDs to work one. */
-  protected final List<byte[]> tsuids;
+  protected List<byte[]> tsuids;
   
   /** The number of get requests batches allowed to be outstanding at
    * any given time. */
-  protected final int concurrency_multi_get;
+  protected int concurrency_multi_get;
   
   /** Whether or not we're fetching rows in reverse. */
-  protected final boolean reversed;
+  protected boolean reversed;
```

Comment:
why?

Reply:
Cause it's a pooled object now so this has to be reset on each query.

Reply:
i mean why use reversed at all.

Reply:
oh cause we will eventually stream data from most recent to oldest so we can load like Splunk does.